llvm/include/llvm/CodeGen/GlobalISel/MachineIRBuilder.h

```diff
@@ -1273,6 +1273,34 @@ class MachineIRBuilder {
                                 const SrcOp &Op0, const SrcOp &Op1,
                                 std::optional<unsigned> Flags = std::nullopt);
 
+  /// Build and insert a \p Res = G_SCMP \p Op0, \p Op1
+  ///
+  /// \pre setBasicBlock or setMI must have been called.
+
+  /// \pre \p Res must be a generic virtual register with scalar or
+  ///      vector type. Typically this starts as s2 or <N x s2>.
+  /// \pre \p Op0 and Op1 must be generic virtual registers with the
+  ///      same number of elements as \p Res. If \p Res is a scalar,
+  ///      \p Op0 must be a scalar.
```

Comment:
If Res is a scalar, does Op0 and Op1 need to have the same bitwidth?

Reply:
Nope. `Op0` and `Op1` need to be the same type, but `Res` could have a different bitwidth as long as they share is the same shape. The same holds true for icmp and fcmp.